packages/webapp/src/components/Filter/FilterPillSelect/index.jsx

```diff
@@ -46,50 +64,79 @@ const FilterPillSelect = ({
+  );
+
+  useEffect(() => {
+    if (shouldReset) {
+      updateAllFilters(false);
     }
-  }, [shouldReset]);
+  }, [shouldReset, updateAllFilters]);
+
+  const selectAllDisabled = useMemo(
+    () => !options.some((option) => !filterState[option.value]?.active),
+    [options, filterState],
+  );
+  const clearAllDisabled = useMemo(
+    () => !options.some((option) => filterState[option.value]?.active),
+    [options, filterState],
+  );
 
   return (
-    <div className={clsx(styles.container, open && styles.openContainer)} style={style}>
+    <>
       <div
-        className={clsx(styles.head, open && styles.openHead, isDisabled && styles.disabled)}
-        onClick={isDisabled ? null : () => setOpen(!open)}
+        className={clsx(styles.container, open && styles.openContainer, className)}
+        style={style}
       >
-        <div>{subject}</div>
-        {counter > 0 && (
-          <>
-            <div className={styles.circle} />
-            <div style={{ flexGrow: '1' }} />
-            <div className={styles.counter}>{`+${counter}`}</div>
-          </>
-        )}
-        <BsChevronDown style={open ? { transform: 'scaleY(-1)' } : {}} />
+        <div
+          className={clsx(styles.head, open && styles.openHead, isDisabled && styles.disabled)}
+          onClick={isDisabled ? null : () => setOpen(!open)}
+        >
+          <div>{subject}</div>
+          {counter > 0 && (
+            <>
+              <div className={styles.circle} />
+              <div style={{ flexGrow: '1' }} />
+              <div className={styles.counter}>{`${counter}`}</div>
+            </>
+          )}
+          <BsChevronDown style={open ? { transform: 'scaleY(-1)' } : {}} />
+        </div>
+        <fieldset className={styles.dropdown} style={{ display: open ? 'flex' : 'none' }}>
+          {options.map((option) => {
+            return (
+              <CheckBoxPill
+                label={option.label}
+                value={option.value}
+                checked={filterState[option.value].active}
+                updateFilterState={updateFilterState}
+                key={option.value}
+                isDisabled={isDisabled}
+              />
+            );
+          })}
+        </fieldset>
       </div>
-      <fieldset className={styles.dropdown} style={{ display: open ? 'flex' : 'none' }}>
-        {options.map((option) => {
-          return (
-            <CheckBoxPill
-              label={option.label}
-              value={option.value}
-              checked={filterState[option.value].active}
-              updateFilterState={updateFilterState}
-              key={option.value}
-              isDisabled={isDisabled}
-            />
-          );
-        })}
-      </fieldset>
-    </div>
+      {showIndividualControls && (
+        <FilterControls
+          updateFilters={updateAllFilters}
+          selectAllDisabled={selectAllDisabled}
+          clearAllDisabled={clearAllDisabled}
+        />
+      )}
+    </>
```

Comment:
This is the bulk of the change in this component, added the ability to display individual controls to clear/select all only in this filter